packages/pwa/app/components/drawer-menu/index.jsx

```diff
@@ -177,7 +177,9 @@ const DrawerMenu = ({isOpen, onClose = noop, onLogoClick = noop, root}) => {
                                                         <SignoutIcon boxSize={5} />
                                                         <Text {...styles.signoutText} as="span">
                                                             {intl.formatMessage({
-                                                                defaultMessage: 'Log out'
+                                                                defaultMessage: 'Log out',
+                                                                description:
+                                                                    'drawer_menu.button.log_out'
```

Comment:
the way we structure these "descriptions" is debatable... if the same word "Log out" appears in 10 different places in the app, does that mean the localization team need to translate it 10 times?

how can we structure/name the messages in a way that is the most efficient for other team to help us translate it?

for example (these are all referring to the same word!):
```
{
drawer_menu.button.log_out: "Log out"
header.log_out: "Log out"
account_page.log_out: "Log out"
}
```

just wanna leave this here before the PR gets 10x larger, and i don't want to see us finding out this is not the pattern we want too late, it will be frustrated to change everything again! So I encourage you to find a good pattern before putting in the real work.

@alexvuong

Reply:
`drawer_menu.button.log_out` doesn't look like a description to me, it looks like an `id`!

are we trying to not use `id` for the sake of not like to using `id`?

Reply:
Hey @kevinxh, I think what you suggested here is to create a single place for localized messages so we don't generate them duplicated into the JSON file, which makes sense to me. However, I think it should have a different ticket on its own.

Reply:
The id is not a required field, and we have agreed to not use it in the first place. Hence, using a description is a way to add context for translators on the message. 

As for the syntax, I tried to follow this [doc](https://confluence.internal.salesforce.com/pages/viewpage.action?spaceKey=ENG&title=L10N%3A+Property+key+naming+rules) to make it easier for them as they have been familiar with the convention, but it is definitely necessary to do so on the description. I am still waiting for some feedback from localization team to discuss about that.

Reply:
I want to question the validity of this statement:

> The id is not a required field, and we have agreed to not use it in the first place.

First of all, what's the reason for us not using `id`? 

My understanding is that we thought it takes time and effort and we don't want to do that to slow down development, we thought we could get away by not adding `id`s.. (pls correct me if there is other reasons) 

The reason turns out to be not true, the evidence is your PR. What this PR is doing, is exactly the opposite of what we have agreed.

What i'm saying is, our previous agreement might be a bad decision because now one of us (you being the unlucky one 😛 ) had to go through the entire code base and basically adds those `id` back!

Reply:
That is a good point. As for the id question, I'd like to loop @vmarta into this conversation on that agreement since he knew the best.

I agree that using ids makes the generated JSON file more neat, readable, and avoids duplication for the same messages (e.g "Log out"). This could potentially become a refactor ticket to add those ids back.

Reply:
Just FYI some conversation can be found on Slack: https://mobify.slack.com/archives/C0DKK1FJS/p1632442212393600

Reply:
hey @alexvuong , we just had a quick meeting with the team and we come to an agreement to use plain english descriptions instead of ID style description. And let's hear what the translator say about that.